From aed238d1566cd8f662606e42eac988422a55c995 Mon Sep 17 00:00:00 2001
From: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Date: Mon, 13 Jan 2020 15:26:27 +0530
Subject: [PATCH 0831/1852] PCI: XDMA PL PCIe: Translate INTx range to hwirqs
 0-3

commit 884bedc65ea2312b40cf9299c560b2b1a835c5df from
https://github.com/Xilinx/linux-xlnx.git

PCI core provided new pci_irqd_intx_xlate() helper to translate
the INTx 1-4 range into the 0-3 range suitable for the IRQ domain of size
4, and stop adding 1 to the hwirq number decoded from the interrupt FIFO
which is already in the range 0-3.

Signed-off-by: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/pci/controller/pcie-xdma-pl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/controller/pcie-xdma-pl.c b/drivers/pci/controller/pcie-xdma-pl.c
index 05511376018b..4c713f4554f6 100644
--- a/drivers/pci/controller/pcie-xdma-pl.c
+++ b/drivers/pci/controller/pcie-xdma-pl.c
@@ -243,6 +243,7 @@ static int xilinx_pcie_intx_map(struct irq_domain *domain, unsigned int irq,
 /* INTx IRQ Domain operations */
 static const struct irq_domain_ops intx_domain_ops = {
 	.map = xilinx_pcie_intx_map,
+	.xlate = pci_irqd_intx_xlate,
 };
 
 /**
@@ -303,7 +304,7 @@ static irqreturn_t xilinx_pcie_intr_handler(int irq, void *data)
 
 		for_each_set_bit(bit, &intr_val, INTX_NUM)
 			generic_handle_irq(irq_find_mapping(port->leg_domain,
-							    bit + 1));
+							    bit));
 	}
 
 	if (status & XILINX_PCIE_INTR_MSI) {
-- 
2.31.1

